webapp/channels/src/components/common/hooks/usePluginVisibilityInSharedChannel.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useSelector} from 'react-redux';
+
+import {getFeatureFlagValue} from 'mattermost-redux/selectors/entities/general';
+
+import type {GlobalState} from 'types/store';
+
+/**
+ * Custom hook to determine if plugin components should be visible in a shared channel.
+ *
+ * @param isSharedChannel - Whether the current channel is a shared channel
+ * @returns true if plugins should be visible, false otherwise
+ *
+ * Plugins are visible when:
+ * - The channel is not shared, OR
+ * - The channel is shared AND the EnableSharedChannelsPlugins feature flag is enabled
+ */
+export function usePluginVisibilityInSharedChannel(isSharedChannel: boolean): boolean {
```

Comment:
It might make this a bit easier to use if it took a channel ID and got the channel from the state itself rather than if it required us to pass in true/false. 0/5 though

Reply:
I like it too. I was a bit hesitant of controlling corner cases inside the hook, but its use is way more direct now from the different components. PTAL https://github.com/mattermost/mattermost/pull/31185/commits/1ff3a36fbbbf63986a111a73a3499c3fe3cf115c